### ISSUE TITLE
Data truncation: Data too long for column 'action_type' at row 1

### ISSUE THREAD

**Issue by danifr** [rundeck/rundeck#1179]
Hi,

after the upgrade to 2.5.0 all my jobs hang after being executed.
If I take a look at the `service.log` file, I see the following error:
```
INFO  ExecutionUtilService: Execution successful: 7381
ERROR SqlExceptionHelper: Data truncation: Data too long for column 'action_type' at row 1
ERROR ExecutionJob: Execution 7381 save result status: caught exception: Hibernate operation: could not execute statement; SQL [n/a]; Data truncation: Data too long for column 'action_type' at row 1; nested exception is com.mysql.jdbc.MysqlDataTruncation: Data truncation: Data too long for column 'action_type' at row 1
org.springframework.dao.DataIntegrityViolationException: Hibernate operation: could not execute statement; SQL [n/a]; Data truncation: Data too long for column 'action_type' at row 1; nested exception is com.mysql.jdbc.MysqlDataTruncation: Data truncation: Data too long for column 'action_type' at row 1
	at rundeck.services.ReportService.reportExecutionResult(ReportService.groovy:49)
	at rundeck.services.ExecutionService.logExecution(ExecutionService.groovy:671)
	at rundeck.services.ExecutionService$_saveExecutionState_closure54.doCall(ExecutionService.groovy:1887)
	at rundeck.services.ExecutionService.saveExecutionState(ExecutionService.groovy:1809)
	at rundeck.quartzjobs.ExecutionJob$_saveState_closure3.doCall(ExecutionJob.groovy:449)
	at rundeck.quartzjobs.ExecutionJob.withRetry(ExecutionJob.groovy:392)
	at rundeck.quartzjobs.ExecutionJob.saveState(ExecutionJob.groovy:458)
	at rundeck.quartzjobs.ExecutionJob.execute_internal(ExecutionJob.groovy:143)
	at rundeck.quartzjobs.ExecutionJob$_execute_closure1.doCall(ExecutionJob.groovy:71)
	at com.codahale.metrics.Timer.time(Timer.java:99)
	at rundeck.quartzjobs.ExecutionJob.execute(ExecutionJob.groovy:70)
	at org.quartz.core.JobRunShell.run(JobRunShell.java:202)
	at org.quartz.simpl.SimpleThreadPool$WorkerThread.run(SimpleThreadPool.java:573)
Caused by: com.mysql.jdbc.MysqlDataTruncation: Data truncation: Data too long for column 'action_type' at row 1
	at com.mysql.jdbc.MysqlIO.checkErrorPacket(MysqlIO.java:3833)
	at com.mysql.jdbc.MysqlIO.checkErrorPacket(MysqlIO.java:3771)
	at com.mysql.jdbc.MysqlIO.sendCommand(MysqlIO.java:2435)
	at com.mysql.jdbc.MysqlIO.sqlQueryDirect(MysqlIO.java:2582)
	at com.mysql.jdbc.ConnectionImpl.execSQL(ConnectionImpl.java:2535)
	at com.mysql.jdbc.PreparedStatement.executeInternal(PreparedStatement.java:1911)
	at com.mysql.jdbc.PreparedStatement.executeUpdate(PreparedStatement.java:2145)
	at com.mysql.jdbc.PreparedStatement.executeUpdate(PreparedStatement.java:2081)
	at com.mysql.jdbc.PreparedStatement.executeUpdate(PreparedStatement.java:2066)
	... 13 more

```

Forgot to mention:
* Rundeck 2.5.0 running on Centos 7.1
* I'm using an external mysql database

Any ideas?
Thanks!

**Comment by danifr** [comment]
Hi again,

I managed to fix this problem.
This is what I did.

```
danifr@centos:~$ mysql -h db-rundeck1 -u rundeckuser -p
```
```
MySQL [(none)]> use rundeck;
MySQL [rundeck]> SHOW TABLES;
+----------------------------+
| Tables_in_rundeck          |
+----------------------------+
| auth_token                 |
| base_report                |
| execution                  |
| log_file_storage_request   |
| node_filter                |
| notification               |
| orchestrator               |
| project                    |
| rdoption                   |
| rdoption_values            |
| rduser                     |
| report_filter              |
| scheduled_execution        |
| scheduled_execution_filter |
| storage                    |
| workflow                   |
| workflow_step              |
| workflow_workflow_step     |
+----------------------------+
18 rows in set (0.00 sec)
MySQL [rundeck]> show columns from base_report;
+-----------------+--------------+------+-----+---------+----------------+
| Field           | Type         | Null | Key | Default | Extra          |
+-----------------+--------------+------+-----+---------+----------------+
| id              | bigint(20)   | NO   | PRI | NULL    | auto_increment |
| version         | bigint(20)   | NO   |     | NULL    |                |
| action_type     | varchar(7)   | NO   |     | NULL    |                |
| author          | varchar(255) | NO   |     | NULL    |                |
| ctx_name        | varchar(255) | YES  |     | NULL    |                |
| ctx_project     | varchar(255) | NO   |     | NULL    |                |
| ctx_type        | varchar(255) | YES  |     | NULL    |                |
| date_completed  | datetime     | NO   |     | NULL    |                |
| date_started    | datetime     | NO   |     | NULL    |                |
| mapref_uri      | varchar(255) | YES  |     | NULL    |                |
| message         | longtext     | NO   |     | NULL    |                |
| node            | varchar(255) | YES  |     | NULL    |                |
| report_id       | varchar(255) | YES  |     | NULL    |                |
| status          | varchar(7)   | NO   |     | NULL    |                |
| tags            | varchar(255) | YES  |     | NULL    |                |
| title           | longtext     | NO   |     | NULL    |                |
| class           | varchar(255) | NO   |     | NULL    |                |
| aborted_by_user | varchar(255) | YES  |     | NULL    |                |
| adhoc_execution | bit(1)       | YES  |     | NULL    |                |
| adhoc_script    | longtext     | YES  |     | NULL    |                |
| ctx_command     | varchar(255) | YES  |     | NULL    |                |
| ctx_controller  | varchar(255) | YES  |     | NULL    |                |
| jc_exec_id      | varchar(255) | YES  |     | NULL    |                |
| jc_job_id       | varchar(255) | YES  |     | NULL    |                |
+-----------------+--------------+------+-----+---------+----------------+
```
Seems that **action_type** and **status** are the guys causing problems.
```
MySQL [rundeck]> ALTER TABLE base_report MODIFY action_type VARCHAR(255);
MySQL [rundeck]> ALTER TABLE base_report MODIFY status VARCHAR(255);
```
I relaunched the job and everything seems to be working fine now.

**Comment by Zophar78** [comment]
Same issue here... after upgrading from 2.1.2 to 2.5.0... Same error... 
After a fresh install of 2.5 base_report table looks like that:
mysql> SHOW columns FROM base_report;
+-----------------+---------------+------+-----+---------+----------------+
| Field           | Type          | Null | Key | Default | Extra          |
+-----------------+---------------+------+-----+---------+----------------+
| id              | bigint(20)    | NO   | PRI | NULL    | auto_increment |
| version         | bigint(20)    | NO   |     | NULL    |                |
| action_type     | varchar(256)  | NO   |     | NULL    |                |
| author          | varchar(255)  | NO   |     | NULL    |                |
| ctx_name        | varchar(255)  | YES  |     | NULL    |                |
| ctx_project     | varchar(255)  | NO   |     | NULL    |                |
| ctx_type        | varchar(255)  | YES  |     | NULL    |                |
| date_completed  | datetime      | NO   |     | NULL    |                |
| date_started    | datetime      | NO   |     | NULL    |                |
| mapref_uri      | varchar(255)  | YES  |     | NULL    |                |
| message         | longtext      | NO   |     | NULL    |                |
| node            | varchar(255)  | YES  |     | NULL    |                |
| report_id       | varchar(3072) | YES  |     | NULL    |                |
| status          | varchar(256)  | NO   |     | NULL    |                |
| tags            | varchar(255)  | YES  |     | NULL    |                |
| title           | longtext      | NO   |     | NULL    |                |
| class           | varchar(255)  | NO   |     | NULL    |                |
| aborted_by_user | varchar(255)  | YES  |     | NULL    |                |
| adhoc_execution | bit(1)        | YES  |     | NULL    |                |
| adhoc_script    | longtext      | YES  |     | NULL    |                |
| ctx_command     | varchar(255)  | YES  |     | NULL    |                |
| ctx_controller  | varchar(255)  | YES  |     | NULL    |                |
| jc_exec_id      | varchar(255)  | YES  |     | NULL    |                |
| jc_job_id       | varchar(255)  | YES  |     | NULL    |                |
+-----------------+---------------+------+-----+---------+----------------+


After 2.5 upgrade (from 2.1.2):
mysql> SHOW columns FROM base_report;
+-----------------+--------------+------+-----+---------+----------------+
| Field           | Type         | Null | Key | Default | Extra          |
+-----------------+--------------+------+-----+---------+----------------+
| id              | bigint(20)   | NO   | PRI | NULL    | auto_increment |
| version         | bigint(20)   | NO   |     | NULL    |                |
| action_type     | varchar(7)   | NO   |     | NULL    |                |
| author          | varchar(255) | NO   |     | NULL    |                |
| ctx_name        | varchar(255) | YES  |     | NULL    |                |
| ctx_project     | varchar(255) | NO   |     | NULL    |                |
| ctx_type        | varchar(255) | YES  |     | NULL    |                |
| date_completed  | datetime     | NO   |     | NULL    |                |
| date_started    | datetime     | NO   |     | NULL    |                |
| mapref_uri      | varchar(255) | YES  |     | NULL    |                |
| message         | longtext     | NO   |     | NULL    |                |
| node            | varchar(255) | YES  |     | NULL    |                |
| report_id       | varchar(255) | YES  |     | NULL    |                |
| status          | varchar(7)   | NO   |     | NULL    |                |
| tags            | varchar(255) | YES  |     | NULL    |                |
| title           | longtext     | NO   |     | NULL    |                |
| class           | varchar(255) | NO   |     | NULL    |                |
| aborted_by_user | varchar(255) | YES  |     | NULL    |                |
| adhoc_execution | bit(1)       | YES  |     | NULL    |                |
| adhoc_script    | longtext     | YES  |     | NULL    |                |
| ctx_command     | varchar(255) | YES  |     | NULL    |                |
| ctx_controller  | varchar(255) | YES  |     | NULL    |                |
| jc_exec_id      | varchar(255) | YES  |     | NULL    |                |
| jc_job_id       | varchar(255) | YES  |     | NULL    |                |
+-----------------+--------------+------+-----+---------+----------------+

**Comment by robgott** [comment]
Just confirming that this issue seems to happen with the default H2 DB not just MySQL.

**Comment by robgott** [comment]
looks like the issue will be the same for the internal H2 DB

FIELD  	|TYPE  	|NULL  	|KEY  	|DEFAULT  
-------------|------------|------------|------------|--------------
ID	|BIGINT(19)	|NO	|PRI	|(NEXT VALUE FOR PUBLIC.SYSTEM_SEQUENCE_2982C3E8_D7E2_4FE1_AB83_A51A27EBAE46)
VERSION	|BIGINT(19)	|NO		|NULL
ACTION_TYPE	|VARCHAR(7)	|NO		|NULL
AUTHOR	|VARCHAR(255)	|NO		|NULL
CTX_NAME	|VARCHAR(255)	|YES		|NULL
CTX_PROJECT	|VARCHAR(255)	|NO		|NULL
CTX_TYPE	|VARCHAR(255)	|YES		|NULL
DATE_COMPLETED	|TIMESTAMP(23)	|NO		|NULL
DATE_STARTED	|TIMESTAMP(23)	|NO		|NULL
MAPREF_URI	|VARCHAR(255)	|YES		|NULL
MESSAGE	|VARCHAR(2147483647)	|NO		|NULL
NODE	|VARCHAR(255)	|YES		|NULL
REPORT_ID	|VARCHAR(255)	|YES		|NULL
STATUS	|VARCHAR(7)	|NO		|NULL
TAGS	|VARCHAR(255)	|YES		|NULL
TITLE	|VARCHAR(2147483647)|	NO		|NULL
CLASS	|VARCHAR(255)	|NO		|NULL
ABORTED_BY_USER	|VARCHAR(255)	|YES		|NULL
ADHOC_EXECUTION	|BOOLEAN(1)	|YES		|NULL
ADHOC_SCRIPT	|VARCHAR(2147483647)	|YES		|NULL
CTX_COMMAND	|VARCHAR(255)	|YES		|NULL
CTX_CONTROLLER	|VARCHAR(255)	|YES		|NULL
JC_EXEC_ID	|VARCHAR(255)	|YES		|NULL
JC_JOB_ID	|VARCHAR(255)	|YES		|NULL

**Comment by gschueler** [comment]
were these databases all upgraded from 2.3.x or earlier originally?

**Comment by gschueler** [comment]
actually i think the varchar(7) was from 2.1 and earlier.

**Comment by Zophar78** [comment]
I had this when upgrading from 2.1.2

**Comment by gschueler** [comment]
repro'd and tested fix for: mysql, h2, upgrading from 2.1.3 to 2.5
